### PR TITLE
Remove Rails 4, Ruby 2.1 and Ruby 2.2 support

### PR DESCRIPTION
**Pull request by feliperenan**
Devise 5 won't support Rails 4.  This PR removes all dependencies that we have in the code related to Rails 4.

Only Rails 5+ will be supported by Devise 5.

### CODE REVIEW

File: test/rails_app/db/migrate/20100401102949_create_tables.rb [hunk 1]
```diff
@@ -1,8 +1,6 @@
 # frozen_string_literal: true
 
-superclass = ActiveRecord::Migration
-# TODO: Inherit from the 5.0 Migration class directly when we drop support for Rails 4.
-superclass = ActiveRecord::Migration[5.0] if superclass.respond_to?(:[])
+superclass = ActiveRecord::Migration[5.0]
 
 class CreateTables < superclass
```
Comment (tegon) [inline test/rails_app/db/migrate/20100401102949_create_tables.rb]:
```suggestion
class CreateTables < ActiveRecord::Migration[5.0]
```

File: lib/devise/models/authenticatable.rb [hunk 2]
```diff
@@ -195,12 +192,7 @@ def devise_mailer
       #
       def send_devise_notification(notification, *args)
         message = devise_mailer.send(notification, self, *args)
-        # Remove once we move to Rails 4.2+ only.
-        if message.respond_to?(:deliver_now)
-          message.deliver_now
```
Comment (rafaelfranca) [inline lib/devise/models/authenticatable.rb]: I think it is the opposite. `deliver_now` is the new method.

File: test/rails_app/db/migrate/20100401102949_create_tables.rb [hunk 3]
```diff
@@ -1,10 +1,8 @@
 # frozen_string_literal: true
 
-superclass = ActiveRecord::Migration
-# TODO: Inherit from the 5.0 Migration class directly when we drop support for Rails 4.
-superclass = ActiveRecord::Migration[5.0] if superclass.respond_to?(:[])
+superclass = ActiveRecord::Migration[5.0]
```
Comment (tegon) [inline test/rails_app/db/migrate/20100401102949_create_tables.rb]:
```suggestion
```

File: .travis.yml [hunk 4]
```diff
@@ -12,25 +10,9 @@ gemfile:
   - Gemfile
   - gemfiles/Gemfile.rails-5.2-rc1
```
Comment (tegon) [inline .travis.yml]: WDYT about renaming this file to `gemfiles/Gemfile.rails-5.2-stable` since it's stable now?

**Comment by feliperenan** [comment]
@tegon @rafaelfranca I think I've done. Just waiting for you review to squash the commits :).

**Pull request merged**